toolkit/src/components/cut/list-item/index.hbs

```diff
@@ -0,0 +1,24 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+}}
+{{#if (or this.href this.route @onClick)}}
+  <Hds::Interactive
```

Comment:
Notice: be aware that this leads to a violation of an accessibility criteria (no nested interactive elements)
https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-nested-interactive.md

@melsumner can give you more detailed explanation, and if you want, I can pair with you and come up with an alternative implementation that doesn't cause this problem